### PR TITLE
fix(agw): Deleted temporary file emm_headers.hpp and moved functions out of extern

### PR DESCRIPTION
**Pull request by pruthvihebbani**
fix(agw):  Deleted temporary file emm_headers.hpp and moved functions out of extern

## Summary

In order to resolve undefined references, function declarations that were causing undefined references were commented out in the original files and added to a temporary file emm_headers.hpp. Now that all the files in mme_app and nas are migrated to C++, emm_headers.hpp is not needed. In this PR emm_headers.hpp is deleted and the function declarations are uncommented.

## Test Plan

Executed the following builds/tests:

1. s1ap integ_tests
2. mme unit test
3. bazel build --config=production //lte/gateway/c/core:agw_of
4. bazel test //lte/gateway/c/core/oai/test/amf:amf_stateless_test
5. bazel test //lte/gateway/c/core/...
6. export FEATURES=mme_oai; make build_oai

### CODE REVIEW

**Review by ajahl — APPROVED** [review]

**Review by LKreutzer — APPROVED** [review]

**Comment by ajahl** [comment]
@pruthvihebbani, can you fix the conflicts?

File: lte/gateway/c/core/oai/tasks/nas/emm/sap/EmmDeregistered.cpp [hunk 1]
```diff
@@ -55,7 +55,6 @@ extern "C" {
 }
 #endif
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/emm/sap/EmmDeregistered.cpp:56]: Space

File: lte/gateway/c/core/oai/tasks/nas/emm/sap/EmmDeregisteredInitiated.cpp [hunk 2]
```diff
@@ -26,7 +26,6 @@ extern "C" {
 }
 #endif
 #include "lte/gateway/c/core/oai/tasks/nas/emm/emm_data.hpp"
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/emm/sap/EmmDeregisteredInitiated.cpp:29]: Space

File: lte/gateway/c/core/oai/test/nas/test_nas_converter.cpp [hunk 3]
```diff
@@ -23,7 +23,6 @@ extern "C" {
 }
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/test/nas/test_nas_converter.cpp:23]: Remove dynamic_memory_check.h from extern C

File: lte/gateway/c/core/oai/tasks/s6a/s6a_fd_iface.cpp [hunk 4]
```diff
@@ -34,7 +36,6 @@ extern "C" {
 #include "lte/gateway/c/core/oai/include/mme_config.hpp"
 #include "lte/gateway/c/core/oai/include/s6a_messages_types.hpp"
 #include "lte/gateway/c/core/oai/tasks/s6a/s6a_messages.hpp"
-#include "lte/gateway/c/core/oai/tasks/s6a/s6a_fd_iface.hpp"
 #include "orc8r/gateway/c/common/service303/MetricsHelpers.hpp"
 
 #include <iostream>
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/s6a/s6a_fd_iface.cpp]: Move these includes above

File: lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme_nas_procedures.hpp [hunk 5]
```diff
@@ -24,6 +24,7 @@
 
 #pragma once
 
+#include "S1ap_S1AP-PDU.h"
 #include "lte/gateway/c/core/common/common_defs.h"
 #include "lte/gateway/c/core/oai/common/common_types.h"
 #include "lte/gateway/c/core/oai/include/mme_app_messages_types.hpp"
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme_nas_procedures.hpp:39]: Include bstrlib.h, s1ap_messages_types.h under extern C

File: lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme_nas_procedures.cpp [hunk 6]
```diff
@@ -22,6 +22,8 @@
   \email: lionel.gauthier@eurecom.fr
 */
 
+#include "lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme_nas_procedures.hpp"
+
 #include <stdio.h>
 #include <stdint.h>
 #include <stdlib.h>
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme_nas_procedures.cpp:29]: Move bstrlib.h file under extern C.

**Review by rsarwad — APPROVED** [review]
LGTM

**Pull request merged**